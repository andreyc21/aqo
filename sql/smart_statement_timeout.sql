SET statement_timeout = 5000; -- [0.8s]

DROP TABLE IF EXISTS a,b CASCADE;
CREATE TABLE a (x int);
INSERT INTO a (x) SELECT mod(ival,10) FROM generate_series(1,1000) As ival;

CREATE TABLE b (y int);
INSERT INTO b (y) SELECT mod(ival + 1,10) FROM generate_series(1,1000) As ival;

CREATE EXTENSION IF NOT EXISTS aqo;
SET aqo.join_threshold = 0;
SET aqo.mode = 'learn';
SET aqo.show_details = 'off';
SET aqo.learn_statement_timeout = 'on';
SET aqo.statement_timeout = 4; -- [0.8s]

SELECT count(y), pg_sleep(3) FROM a,b where x > 2 * (select min(x) from A,B where x = y);
select * from aqo_queries where query_hash <> 0;
SELECT count(y), pg_sleep(3) FROM a,b where x > 3 * (select min(x) from A,B where x = y);
select * from aqo_queries where query_hash <> 0;
SELECT count(y), pg_sleep(3) FROM a,b where x > 3 * (select min(x) from A,B where x = y);
select * from aqo_queries where query_hash <> 0;
SELECT count(y), pg_sleep(3) FROM a,b where x > (select min(x) from A,B where x = y);
select * from aqo_queries where query_hash <> 0;
SELECT count(y), pg_sleep(3) FROM a,b where x > (select min(x) from A,B where x = y);
select * from aqo_queries where query_hash <> 0;
SELECT count(y), pg_sleep(3) FROM a,b where x > (select min(x) from A,B where x = y);
select * from aqo_queries where query_hash <> 0;
DROP EXTENSION aqo;
