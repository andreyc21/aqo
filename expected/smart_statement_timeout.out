SET statement_timeout = 5000; -- [0.8s]
DROP TABLE IF EXISTS a,b CASCADE;
NOTICE:  table "a" does not exist, skipping
NOTICE:  table "b" does not exist, skipping
CREATE TABLE a (x int);
INSERT INTO a (x) SELECT mod(ival,10) FROM generate_series(1,1000) As ival;
CREATE TABLE b (y int);
INSERT INTO b (y) SELECT mod(ival + 1,10) FROM generate_series(1,1000) As ival;
CREATE EXTENSION IF NOT EXISTS aqo;
SET aqo.join_threshold = 0;
SET aqo.mode = 'learn';
SET aqo.show_details = 'off';
SET aqo.learn_statement_timeout = 'on';
SET aqo.statement_timeout = 4; -- [0.8s]
SELECT count(y), pg_sleep(3) FROM a,b where x > 2 * (select min(x) from A,B where x = y);
NOTICE:  [AQO] Time limit for execution of the statement was expired. AQO tried to learn on partial data. Current timeout is 4, mean integral error is 0.000000
 count  | pg_sleep 
--------+----------
 900000 | 
(1 row)

select * from aqo_queries where query_hash <> 0;
    query_hash     | learn_aqo | use_aqo |    fspace_hash    | auto_tuning | flex_timeout | count_increase_timeout 
-------------------+-----------+---------+-------------------+-------------+--------------+------------------------
 42929091640608990 | t         | t       | 42929091640608990 | f           |            5 |                      1
(1 row)

SELECT count(y), pg_sleep(3) FROM a,b where x > 3 * (select min(x) from A,B where x = y);
NOTICE:  [AQO] Time limit for execution of the statement was expired. AQO tried to learn on partial data. Current timeout is 4, mean integral error is 0.866297
 count  | pg_sleep 
--------+----------
 900000 | 
(1 row)

select * from aqo_queries where query_hash <> 0;
    query_hash     | learn_aqo | use_aqo |    fspace_hash    | auto_tuning | flex_timeout | count_increase_timeout 
-------------------+-----------+---------+-------------------+-------------+--------------+------------------------
 42929091640608990 | t         | t       | 42929091640608990 | f           |            5 |                      1
(1 row)

SELECT count(y), pg_sleep(3) FROM a,b where x > 3 * (select min(x) from A,B where x = y);
NOTICE:  [AQO] Time limit for execution of the statement was expired. AQO tried to learn on partial data. Current timeout is 4, mean integral error is 0.435709
 count  | pg_sleep 
--------+----------
 900000 | 
(1 row)

select * from aqo_queries where query_hash <> 0;
    query_hash     | learn_aqo | use_aqo |    fspace_hash    | auto_tuning | flex_timeout | count_increase_timeout 
-------------------+-----------+---------+-------------------+-------------+--------------+------------------------
 42929091640608990 | t         | t       | 42929091640608990 | f           |            5 |                      1
(1 row)

SELECT count(y), pg_sleep(3) FROM a,b where x > (select min(x) from A,B where x = y);
NOTICE:  [AQO] Time limit for execution of the statement was expired. AQO tried to learn on partial data. Current timeout is 4, mean integral error is 0.000000
 count  | pg_sleep 
--------+----------
 900000 | 
(1 row)

select * from aqo_queries where query_hash <> 0;
     query_hash      | learn_aqo | use_aqo |     fspace_hash     | auto_tuning | flex_timeout | count_increase_timeout 
---------------------+-----------+---------+---------------------+-------------+--------------+------------------------
   42929091640608990 | t         | t       |   42929091640608990 | f           |            5 |                      1
 6582036157989373325 | t         | t       | 6582036157989373325 | f           |           20 |                      2
(2 rows)

SELECT count(y), pg_sleep(3) FROM a,b where x > (select min(x) from A,B where x = y);
NOTICE:  [AQO] Time limit for execution of the statement was expired. AQO tried to learn on partial data. Current timeout is 4, mean integral error is 0.866297
 count  | pg_sleep 
--------+----------
 900000 | 
(1 row)

select * from aqo_queries where query_hash <> 0;
     query_hash      | learn_aqo | use_aqo |     fspace_hash     | auto_tuning | flex_timeout | count_increase_timeout 
---------------------+-----------+---------+---------------------+-------------+--------------+------------------------
   42929091640608990 | t         | t       |   42929091640608990 | f           |            5 |                      1
 6582036157989373325 | t         | t       | 6582036157989373325 | f           |           20 |                      2
(2 rows)

SELECT count(y), pg_sleep(3) FROM a,b where x > (select min(x) from A,B where x = y);
NOTICE:  [AQO] Time limit for execution of the statement was expired. AQO tried to learn on partial data. Current timeout is 4, mean integral error is 0.435709
 count  | pg_sleep 
--------+----------
 900000 | 
(1 row)

select * from aqo_queries where query_hash <> 0;
     query_hash      | learn_aqo | use_aqo |     fspace_hash     | auto_tuning | flex_timeout | count_increase_timeout 
---------------------+-----------+---------+---------------------+-------------+--------------+------------------------
   42929091640608990 | t         | t       |   42929091640608990 | f           |            5 |                      1
 6582036157989373325 | t         | t       | 6582036157989373325 | f           |           20 |                      2
(2 rows)

DROP EXTENSION aqo;
